google-cloud-tools-plugin/src/com/google/gct/idea/appengine/cloud/ManagedVmCloudType.java

```diff
@@ -196,6 +204,56 @@ public String getPresentableName() {
       public void setName(String name) {
         this.name = name;
       }
+
+      @NotNull
+      @Override
+      public DeploymentSourceType<?> getType() {
+        return DeploymentSourceType.EP_NAME.findExtension(UserSpecifiedPathDeploymentSourceType.class);
+      }
+    }
+  }
+
+  public static class UserSpecifiedPathDeploymentSourceType extends
+      DeploymentSourceType<ModuleDeploymentSource> {
+    private static final String NAME_ATTRIBUTE = "name";
+
+    public UserSpecifiedPathDeploymentSourceType() {
+      super("filesystem WAR/JAR module");
+    }
+
+    @NotNull
+    @Override
+    /**
+     * Restore presentable name (e.g., to be "Filesystem JAR or WAR file - <file path>") of
+     * UserSpecifiedPathDeploymentSource.
+     */
+    public ModuleDeploymentSource load(@NotNull Element tag, @NotNull Project project) {
+      UserSpecifiedPathDeploymentSource userSpecifiedSource = new UserSpecifiedPathDeploymentSource(
+          ModulePointerManager.getInstance(project).create(tag.getAttributeValue(NAME_ATTRIBUTE)));
+
+      Element settings = tag.getChild(DeployToServerRunConfiguration.SETTINGS_ELEMENT);
+      if (settings != null) {
+        String filePath = settings.getAttributeValue("userSpecifiedArtifactPath");
```

Comment:
"userSpecifiedArtifactPath" is an attribute you can see in ManagedVmDeploymentConfiguration. It has to match the attribute name defined there. Does it still make sense to use properties?